tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -25,19 +25,20 @@ public SingleSaveBlobStoreDAO(BlobStoreDAO blobStoreDAO,
 
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
         return Mono.from(blobIdList.isStored(blobId))
-                .filter(isStored -> Objects.equals(isStored, false))
-                .switchIfEmpty(Mono.error(new ObjectStoreException("Can not save duplicate blob when single save is enabled")))
-                .then(Mono.from(blobStoreDAO.save(bucketName, blobId, data))
-                        .then(Mono.from(blobIdList.store(blobId))))
+                .filter(isStored -> isStored.equals(false))
+                .switchIfEmpty(Mono.empty())
+                .then(Mono.from(blobIdList.store(blobId))
+                        .then(Mono.from(blobStoreDAO.save(bucketName, blobId, data)))
+                        .onErrorResume(error -> Mono.from(blobIdList.remove(blobId)).then()))
```

Comment:
```suggestion
                .filter(isStored -> isStored.equals(false))
                .switchIfEmpty((Mono.from(blobStoreDAO.save(bucketName, blobId, data))
                        .then(Mono.from(blobIdList.store(blobId))))
```

 - Save first in the blobStore
 - Then to the blobId list
 
 Yes, we might end up storing things two times, but it is today behaviour and it protect us against inconsistencies.